src/shared/lib/markDownRules.tsx

```diff
@@ -547,7 +575,52 @@ const getContainerAlignTag = (parents: ASTNode[]): AlignmentTag | undefined => {
 };
 
 export const preprocessImageLinks = (content: string) => {
-  return content?.replace(/(!\[.*\]\s*\(.*?) =\d*x\d*(\))/g, '$1$2');
+  const regexSize = /\s*=\s*(\d+)x(\d+)/g;
+  const regexImage = /!\[.*?\]\((.*?)\)/g;
+
+  const getImageSize = (url: string) => {
+    return new Promise<{ width: number; height: number }>((resolve, reject) => {
+      Image.getSize(
+        url,
+        (width, height) => resolve({ width, height }),
+        error => reject(error),
+      );
+    });
+  };
+
+  const matches = [...content.matchAll(regexImage)];
+
+  const promises = matches.map(match => {
```

Comment:
I think you could drastically improve readability of the peace of code by re-writing the `preprocessImageLinks` to async/await  style instead of promises.
This way, you will get rid of the `promises ` variable (and the subsequent call of `Promise.all`) and `Promise.resolve` calls